src/main/java/jenkins/branch/Branch.java

```diff
@@ -42,6 +42,7 @@
 
 /**
  * A source code branch.
+ * Not to be subclassed outside this plugin.
```

Comment:
It is subclassed in `YamlProjectFactoryTest`, but I think this is indeed possible using `@Restricted`, will try it.